src/sys/linux.rs

```diff
@@ -78,19 +78,46 @@ impl<'a, S: AsRawFd> MptcpSocketRef<'a, S> {
     pub fn has_fallback(&self) -> bool {
         const SOL_MPTCP: libc::c_int = 0x11c;
         const MPTCP_INFO: libc::c_int = 0x1;
+        const MPTCP_INFO_FLAG_FALLBACK: u32 = 0x1 << 0;
 
         if !has_mptcp_info() {
             return !self.is_mptcp_socket();
```

Comment:
> As `MptcpSocketRef` hold a ref to a `RawFd` (ie. a struct that own a file descriptor) I am not sure whether we can infer whether the socket was a client socket or a server socket.

I see, that makes sense. So a solution here could be to add a new dedicated helper, so users of the lib could use it if they know they are in the right conditions. But maybe this optimisation is not that worth it for the moment because this second option might be confusing.

> I could adapt the `MptcpExt` trait that is available for all `RawFd` to be specific to each `TcpStream` and `TcpListener`, this way I could indicate whether it is a server or client socket but I am usure it is worth it to spare a `getsockopt` only when retrieving `mptcp_status`.

Even if we know if it is a client or a server, we would need to know if the user of the lib is doing that just after the `accept()` or if it wants to know if a fallback happened later.